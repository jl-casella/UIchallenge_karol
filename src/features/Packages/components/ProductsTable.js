import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { Product } from '../types'
import Table from 'atomics/Table'

const ProductsTableRow = styled.tr`
  cursor: pointer;
`

const LocationHeaderCell = styled.th`
  width: 100px;
`

const QuantityHeaderCell = styled.th`
  text-align: right;
  width: 50px;
`

const QuantityCell = styled.td`
  text-align: right;
`

const ProductsTable = ({ items, onProductRowClick }) => (
  <Table>
    <thead>
      <tr>
        <th>SKU</th>
        <LocationHeaderCell>Location</LocationHeaderCell>
        <QuantityHeaderCell>Quantity</QuantityHeaderCell>
      </tr>
    </thead>
    <tbody>
      {items.map(i => {
        return (
          <ProductsTableRow key={i.id} onClick={() => onProductRowClick(i)}>
            <td>{i.sku}</td>
            <td>{i.location}</td>
            <QuantityCell>{i.quantity}</QuantityCell>
          </ProductsTableRow>
        )
      })}
    </tbody>
  </Table>
)

ProductsTable.propTypes = {
  items: PropTypes.arrayOf(Product).isRequired,
  onProductRowClick: PropTypes.func.isRequired
}

export default ProductsTable
