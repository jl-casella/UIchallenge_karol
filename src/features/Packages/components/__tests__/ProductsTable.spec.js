import { noop } from 'lodash'
import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import ProductsTable from '../ProductsTable'

describe('ProductsTable component test', () => {
  const productsMock = [
    {
      id: 1,
      quantity: 5,
      sku: 'green-ball',
      location: 'a1'
    },
    {
      id: 2,
      quantity: 6,
      sku: 'red-ball',
      location: 'a2'
    }
  ]

  it('Should render correctly', () => {
    const { container } = render(
      <ProductsTable items={productsMock} onProductRowClick={noop} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Calls onRowClick handler after clicking on the row', () => {
    const spy = jest.fn()
    const { getByText } = render(
      <ProductsTable items={productsMock} onProductRowClick={spy} />
    )

    const skuCell = getByText('green-ball')
    const locationCell = getByText('a1')
    const quantityCell = getByText('5')
    expect(spy).toHaveBeenCalledTimes(0)

    fireEvent.click(skuCell)
    fireEvent.click(locationCell)
    fireEvent.click(quantityCell)
    expect(spy).toHaveBeenCalledTimes(3)
  })
})
