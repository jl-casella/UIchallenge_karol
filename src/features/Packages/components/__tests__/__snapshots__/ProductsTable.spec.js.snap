// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsTable component test Should render correctly 1`] = `
<table
  class="Table-nnk0cv-0 ecmwxU"
>
  <thead>
    <tr>
      <th>
        SKU
      </th>
      <th
        class="ProductsTable__LocationHeaderCell-sc-1vgup2v-1 gxVAmz"
      >
        Location
      </th>
      <th
        class="ProductsTable__QuantityHeaderCell-sc-1vgup2v-2 gRyJwF"
      >
        Quantity
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      class="ProductsTable__ProductsTableRow-sc-1vgup2v-0 gZpYFG"
    >
      <td>
        green-ball
      </td>
      <td>
        a1
      </td>
      <td
        class="ProductsTable__QuantityCell-sc-1vgup2v-3 hRppQA"
      >
        5
      </td>
    </tr>
    <tr
      class="ProductsTable__ProductsTableRow-sc-1vgup2v-0 gZpYFG"
    >
      <td>
        red-ball
      </td>
      <td>
        a2
      </td>
      <td
        class="ProductsTable__QuantityCell-sc-1vgup2v-3 hRppQA"
      >
        6
      </td>
    </tr>
  </tbody>
</table>
`;
